coderd/coderd.go

```diff
@@ -1528,11 +1529,28 @@ func New(options *Options) *API {
 		})
 		r.Route("/insights", func(r chi.Router) {
 			r.Use(apiKeyMiddleware)
-			r.Get("/daus", api.deploymentDAUs)
-			r.Get("/user-activity", api.insightsUserActivity)
+			r.Group(func(r chi.Router) {
+				r.Use(
+					func(next http.Handler) http.Handler {
+						return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+							if options.DeploymentValues.DisableTemplateInsights.Value() {
+								httpapi.Write(context.Background(), rw, http.StatusForbidden, codersdk.Response{
+									Message: "Forbidden.",
```

Comment:
IMO 404 is more appropriate here, forbidden could indicate that it's there but you're just not allowed to see it.

Reply:
Agreed, that makes sense.